src/main/java/de/tum/in/www1/artemis/service/ProgrammingExerciseExportService.java

```diff
@@ -149,6 +157,115 @@ private void deleteTempZipRepoFiles(List<Path> pathsToZipeedRepos) {
         }
     }
 
+    /**
+     * downloads all repos of the exercise and runs JPlag
+     *
+     * @param programmingExerciseId the id of the programming exercises which should be checked
+     * @return a zip file that can be returned to the client
+     * @throws ExitException is thrown if JPlag exits unexpectedly
+     * @throws IOException is thrown for file handling errors
+     */
+    public File checkPlagiarism(long programmingExerciseId) throws ExitException, IOException {
```

Comment:
Codacy found an issue: [The method 'checkPlagiarism(long)' has an NPath complexity of 864, current threshold is 200](https://app.codacy.com/gh/ls1intum/Artemis/pullRequest?prid=5968852)